app/models/katello/glue/candlepin/pool.rb

```diff
@@ -169,7 +171,7 @@ def create_activation_key_associations
       end
 
       def hypervisor
-        ::Katello::Host::SubscriptionFacet.find_by(:uuid => host_id).try(:host) if host_id
+        ::Host.find(self.hypervisor_id) if self.hypervisor_id
```

Comment:
i would use find_by(:id => ID) to fail gracefully

Reply:
good catch, updated